shared/actions/chat.js

```diff
@@ -907,46 +975,58 @@ function * _setupChatHandlers (): SagaGenerator<any, any> {
 
 const inboxSelector = (state: TypedState, conversationIDKey) => state.chat.get('inbox')
 
-function * selectValidConversation () {
+function * _ensureValidSelectedChat (onlyIfNoSelection: boolean) {
+  // yield call(delay, 1) // need this as we want the reducer etc to run first before we deal with this. TODO talk about this
+
   if (isMobile) {
     return // Mobile doens't auto select a conversation
   }
   const inbox = yield select(inboxSelector)
   if (inbox.count()) {
     const conversationIDKey = yield select(_selectedSelector)
-    // Is the currently selected one not validated? or empty
-    if (!inbox.find(c => c.get('conversationIDKey') === conversationIDKey &&
-          c.get('validated') && (!c.get('isEmpty') || c.get('youCreated')))) {
-      const validInbox = inbox.find(i => i.get('validated') && (!i.get('isEmpty') || i.get('youCreated')))
-      if (validInbox) {
-        const validInboxConvIDKey = validInbox.get('conversationIDKey')
-        yield put(selectConversation(validInboxConvIDKey, false))
-        yield put(loadMoreMessages(validInboxConvIDKey, true))
-      } else {
-        yield put(selectConversation(null, false))
-      }
+
+    if (onlyIfNoSelection && conversationIDKey) {
+      return
+    }
+
+    const alwaysShow = yield select(_alwaysShowSelector)
+
+    const current = inbox.find(c => c.get('conversationIDKey') === conversationIDKey)
+    // current is good
+    if (current && (!current.get('isEmpty') || alwaysShow.has(conversationIDKey))) {
+      return
+    }
+
+    const firstGood = inbox.find(i => !i.get('isEmpty') || alwaysShow.has(i.get('conversationIDKey')))
+    if (firstGood) {
+      const conversationIDKey = firstGood.get('conversationIDKey')
+      yield put(selectConversation(conversationIDKey, false))
+      // yield put(loadMoreMessages(validInboxConvIDKey, true))
     } else {
-      yield put(loadMoreMessages(conversationIDKey, true))
+      yield put(selectConversation(null, false))
     }
   }
 }
 
 const followingSelector = (state: TypedState) => state.config.following
 
-function * _loadInbox (action: ?LoadInbox): SagaGenerator<any, any> {
+function * _loadInbox (action: LoadInbox): SagaGenerator<any, any> {
   const channelConfig = singleFixedChannelConfig([
     'chat.1.chatUi.chatInboxUnverified',
     'chat.1.chatUi.chatInboxConversation',
     'chat.1.chatUi.chatInboxFailed',
     'finished',
   ])
 
+  const singleLoad = action && !!action.payload.onlyLoad
```

Comment:
if we're only loading a single item, don't do the normal actions which mutate the inbox as a whole